lib/internal/Magento/Framework/Mview/View/Subscription.php

```
@@ -193,11 +191,23 @@ protected function getLinkedViews()
      * Build trigger statement for INSERT, UPDATE, DELETE events
      *
      * @param string $event
-     * @param \Magento\Framework\Mview\View\ChangelogInterface $changelog
+     * @param \Magento\Framework\Mview\ViewInterface $view
      * @return string
      */
-    protected function buildStatement($event, $changelog)
+    protected function buildStatement($event, $view)
     {
+        // Get the subscription for the specific view and specific table.
+        // We will use column name from it.
+        $subscriptions = $view->getSubscriptions() ?? [];
+        if (empty($subscriptions[$this->getTableName()])) {
+            return '';
```

Wouldn't be better to throw an exception here instead?